packages/amplify_api/lib/method_channel_api.dart

```diff
@@ -107,7 +107,7 @@ class AmplifyAPIMethodChannel extends AmplifyAPI {
 
     StreamSubscription _subscription = filteredStream.listen((event) {
       if (event['type'] == 'DONE') {
-        onDone();
+        onDone!();
```

Comment:
Isn't `onDone` nullable? If it's not provided, this would cause an issue right? I think we could either initialize `onDone` with some kind of noop if one is not provided or we need to perform a null check here

Reply:
ah good call, added null check